packages/unmock-core/src/nock.ts

```diff
@@ -83,8 +114,59 @@ export type ExtendedValueType =
   | IExtendedObjectType
   | JSONArray
   | JSONObject;
-export interface IExtendedObjectType
+
+export interface ITameExtendedObjectType
   extends Record<string, ExtendedValueType> {} // Defined as interface due to circular reasons
+
+export interface IExtendedObjectType
+  extends Record<string, ExtendedValueType | IMaybeJSONValue> {} // Defined as interface due to circular reasons
+
+const rejectOptionals = (i: IExtendedObjectType): ITameExtendedObjectType =>
+  new Iso<
+    IExtendedObjectType,
+    Array<[string, ExtendedValueType | IMaybeJSONValue]>
+  >(
+    a => Object.entries(a),
+    q => q.reduce((a, b) => ({ ...a, [b[0]]: b[1] }), {}),
+  )
+    .composeTraversal(fromTraversable(array)())
+    .composePrism(
+      new Prism<
+        [string, ExtendedValueType | IMaybeJSONValue],
+        [string, ExtendedValueType]
+      >(
+        a =>
+          ((
+            q: string,
+            r: ExtendedValueType | IMaybeJSONValue,
+          ): Option<[string, ExtendedValueType]> =>
+            MaybeJSONValue.is(r) ? none : some([q, r]))(a[0], a[1]),
+        a => a,
+      ),
+    )
+    .composeGetter(identityGetter())
+    .getAll(i)
+    .reduce((a, b) => ({ ...a, [b[0]]: b[1] }), {});
```

Comment:
Would `asFold()` + folding be more idiomatic `monocle-ts` than `getAll(i)` +  `reduce`? Not important, just a comment.

Reply:
I need to learn more about folds...